src/test/regress/sql/with_clause.sql

```diff
@@ -358,3 +358,63 @@ WITH RECURSIVE r1 AS (
 	)
 )
 SELECT * FROM r1 LIMIT 1;
+--
+-- Test various SELECT statements over DML operations
+--
+create table with_dml(i integer, j integer) distributed by(i);
+create table with_dml_dr(i integer, j integer) distributed replicated;
+-- Test select over insert queries. Gather motion should be there, no matter
+-- what type of distribution is used. Before fix, motion was missed for
+-- distributed replicated table, which caused count(*) to return zero rows.
+explain (costs off)
+with cte as (
+    insert into with_dml select i, i * 100 from generate_series(1,5) i
+    returning i
+) select count(*) from cte;
```

Comment:
You might add `where` clause for `select` stmt to verify whether it pushes down to `insert` one.